app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/masterpasswordgenerator/MasterPasswordGeneratorScreen.kt

```diff
@@ -0,0 +1,231 @@
+        SnackbarHostState()
+    }
+    EventsEffect(viewModel = viewModel) { event ->
+        when (event) {
+            MasterPasswordGeneratorEvent.NavigateBack -> onNavigateBack()
+            MasterPasswordGeneratorEvent.NavigateToPreventLockout -> onNavigateToPreventLockout()
+            is MasterPasswordGeneratorEvent.ShowSnackbar -> {
+                snackbarHostState.showSnackbar(
+                    message = event.text.toString(resources),
+                    duration = SnackbarDuration.Short,
+                )
+            }
+        }
+    }
+
+    val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior(rememberTopAppBarState())
+    BitwardenScaffold(
+        modifier = Modifier
+            .fillMaxSize()
+            .nestedScroll(scrollBehavior.nestedScrollConnection),
+        topBar = {
+            MasterPasswordGeneratorTopBar(
+                scrollBehavior = scrollBehavior,
+                onBackClick = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(MasterPasswordGeneratorAction.BackClickAction)
+                    }
+                },
+                onSaveClick = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGeneratorAction.SavePasswordClickAction,
+                        )
+                    }
+                },
+            )
+        },
+        snackbarHost = {
+            SnackbarHost(hostState = snackbarHostState)
+        },
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .padding(innerPadding),
+        ) {
+            MasterPasswordGeneratorContent(
+                generatedPassword = state.generatedPassword,
+                onGenerateNewPassword = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGeneratorAction.GeneratePasswordClickAction,
+                        )
+                    }
+                },
+                onLearnToPreventLockout = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGeneratorAction.PreventLockoutClickAction,
+                        )
+                    }
+                },
+                modifier = Modifier.padding(horizontal = 16.dp),
+            )
+            Spacer(modifier = Modifier.navigationBarsPadding())
+        }
+    }
+}
+
+@Composable
+private fun MasterPasswordGeneratorContent(
+    generatedPassword: String,
+    onGenerateNewPassword: () -> Unit,
+    onLearnToPreventLockout: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Text(
+            text = stringResource(R.string.generate_master_password),
```

Comment:
Should this be in the toolbar?

Reply:
moved this text to the existing toolbar, created story to handle overflow tool bar https://bitwarden.atlassian.net/browse/PM-10851